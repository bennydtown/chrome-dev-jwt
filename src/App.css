.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 100px;
}

.App-header {
  background-color: #e04f5f;
  height: 80px;
  color: white;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
