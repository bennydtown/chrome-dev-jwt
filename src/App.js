import React, { Component } from 'react';
import './App.css';
import Form from './form';

const logoSvgPaths = [
  `M59.8,29.1L44.9,44c-1.3-0.5-2.6-0.7-4-0.7c-6.5,0-11.8,5.3-11.8,11.8s5.3,11.8,11.8,11.8
		c6.5,0,11.8-5.3,11.8-11.8c0-1.4-0.3-2.8-0.7-4l0.7-0.7v-4.7h4.7v-4.7h4.7v-4.7h4.7v-7.1H59.8z M43.6,60.5c-0.1,0.6-0.5,1-1.1,1.2
		c-2.1,0.8-4.7,0.4-6.4-1.3c-1.7-1.7-2.2-4.3-1.3-6.4c0.2-0.5,0.7-0.9,1.2-1c0.6-0.1,1.1,0.1,1.6,0.5l5.5,5.5
		C43.6,59.4,43.7,60,43.6,60.5z`,
  `M66.9,29.1h-7.1L44.9,44c-1.3-0.5-2.6-0.7-4-0.7c-6.5,0-11.8,5.3-11.8,11.8
		c0,3.4,1.4,6.4,3.7,8.6l3.3-3.3c-1.7-1.7-2.1-4.2-1.3-6.3c0.2-0.5,0.7-0.9,1.2-1c0.6-0.1,1.1,0.1,1.6,0.5l2.7,2.7l26.6-26.6V29.1z`,
];

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <div style={{ width: '80px', height: '80px' }}>
            <svg className="logo-img logo-react"
            viewBox="10 10 80 80"
            role="img" xmlns="http://www.w3.org/2000/svg">
              <title>Dev JWT</title>
              <path
                className="vt-logomark vt-logo__mark--left"
                fill="white"
                d={logoSvgPaths[0]}
              />
              <path
                className="vt-logomark vt-logo__mark--right"
                fill="white"
                d={logoSvgPaths[1]}
              />
            </svg>
          </div>
          <h1 style={{ float: 'right', fontSize: '28px' }} className="App-title">Dev JWT</h1>
        </header>
        <div>
          Create an unsigned JWT for development.
          <Form />
        </div>
      </div>
    );
  }
}

export default App;
