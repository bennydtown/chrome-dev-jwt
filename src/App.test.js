import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing and no unexpected changes', () => {
  const component = renderer.create(<App />);
  expect(component.toJSON()).toMatchSnapshot();
});
