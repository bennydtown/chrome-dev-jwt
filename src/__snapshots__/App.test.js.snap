// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and no unexpected changes 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      style={
        Object {
          "height": "80px",
          "width": "80px",
        }
      }
    >
      <svg
        className="logo-img logo-react"
        role="img"
        viewBox="10 10 80 80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Dev JWT
        </title>
        <path
          className="vt-logomark vt-logo__mark--left"
          d="M59.8,29.1L44.9,44c-1.3-0.5-2.6-0.7-4-0.7c-6.5,0-11.8,5.3-11.8,11.8s5.3,11.8,11.8,11.8
        		c6.5,0,11.8-5.3,11.8-11.8c0-1.4-0.3-2.8-0.7-4l0.7-0.7v-4.7h4.7v-4.7h4.7v-4.7h4.7v-7.1H59.8z M43.6,60.5c-0.1,0.6-0.5,1-1.1,1.2
        		c-2.1,0.8-4.7,0.4-6.4-1.3c-1.7-1.7-2.2-4.3-1.3-6.4c0.2-0.5,0.7-0.9,1.2-1c0.6-0.1,1.1,0.1,1.6,0.5l5.5,5.5
        		C43.6,59.4,43.7,60,43.6,60.5z"
          fill="white"
        />
        <path
          className="vt-logomark vt-logo__mark--right"
          d="M66.9,29.1h-7.1L44.9,44c-1.3-0.5-2.6-0.7-4-0.7c-6.5,0-11.8,5.3-11.8,11.8
        		c0,3.4,1.4,6.4,3.7,8.6l3.3-3.3c-1.7-1.7-2.1-4.2-1.3-6.3c0.2-0.5,0.7-0.9,1.2-1c0.6-0.1,1.1,0.1,1.6,0.5l2.7,2.7l26.6-26.6V29.1z"
          fill="white"
        />
      </svg>
    </div>
    <h1
      className="App-title"
      style={
        Object {
          "float": "right",
          "fontSize": "28px",
        }
      }
    >
      Dev JWT
    </h1>
  </header>
  <div>
    Create an unsigned JWT for development.
    <div>
      <form
        onSubmit={[Function]}
        style={
          Object {
            "marginRight": "10px",
            "textAlign": "right",
          }
        }
      >
        <div
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <label>
            UserID:
            <input
              name="userId"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                  "width": "280px",
                }
              }
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <label>
            Roles:
            <input
              name="roles"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                  "width": "280px",
                }
              }
              type="text"
              value="[\\"role1\\", \\"role2\\", \\"role3\\"]"
            />
          </label>
        </div>
        <div
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <label>
            Save Cookie to Domain:
            <select
              name="domain"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                  "width": "287px",
                }
              }
            >
              <option
                value="localhost"
              >
                localhost
              </option>
              <option
                value="yourdomain.com"
              >
                yourdomain.com
              </option>
              <option
                value="www.yourdomain.com"
              >
                www.yourdomain.com
              </option>
              <option
                value="dev"
              >
                dev
              </option>
              <option
                value="local"
              >
                local
              </option>
            </select>
          </label>
        </div>
        <input
          style={
            Object {
              "marginTop": "10px",
            }
          }
          type="submit"
          value="Create Token"
        />
      </form>
    </div>
  </div>
</div>
`;
